programs/stake/src/stake_state.rs

```diff
@@ -842,6 +817,83 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+/// After calling `validate_delegate_amount()`, this struct contains calculated values that are used
+/// by the caller.
+struct ValidatedDelegateInfo {
+    stake_amount: u64,
+}
+
+/// Ensure the stake delegation amount is valid.  This checks that the account meets the minimum
+/// balance requirements to delegate stake.  If not, return an error.
+fn validate_delegate_amount(
+    account: &KeyedAccount,
+    meta: &Meta,
+) -> Result<ValidatedDelegateInfo, InstructionError> {
+    let stake_amount = account.lamports()?.saturating_sub(meta.rent_exempt_reserve); // can't stake the rent ;)
```

Comment:
```suggestion
    let stake_amount = account.lamports()?.saturating_sub(meta.rent_exempt_reserve); // can't stake the rent
```
Not sure that winky face is communicating anything useful anymore